### DISCUSSION TITLE
Outline box/shading for content tabs

### DISCUSSION THREAD

**Discussion by EDIflyer** [squidfunk/mkdocs-material#4157] [Q&A] [answered]
I really like the [content tab](https://squidfunk.github.io/mkdocs-material/reference/content-tabs) functionality however if I dont then put it within an admonition or perhaps just a simple code block on each tab I find it difficult when reading to see which content is part of a tab and which is part of the main text (see examples below).

Is there some way to create a box around the tab contents to make it easier to see what is tabbed content vs main text?

**Tab 1 active**
![image](https://user-images.githubusercontent.com/13610277/180603896-62dfbf24-48e6-42d9-a094-33d93d711147.png)

**Tab 2 active**
![image](https://user-images.githubusercontent.com/13610277/180603898-5637e9b7-82d8-48d9-9b31-7fd417b98b52.png)

**Comment by squidfunk** [comment]
You can add a `border` or `box-shadow` with some [additional CSS](https://squidfunk.github.io/mkdocs-material/customization/#additional-css). Material for MkDocs is very extensible and can be easily adjusted to taste. I'm short on time right now, but maybe somebody else can share some customizations!

**Reply by EDIflyer** [reply]
Thanks @squidfunk - haven't play around with CSS too much before, will try and get my head round it 👍

**Comment by rmscode** [comment]
@EDIflyer 

Sorry for being late to the party, but I'm curious to know what you ended up doing. I have found myself wanting to do the same thing.

**Reply by EDIflyer** [reply]
Must confess I never got round to dealing with it! I think it would definitely look nicer with a subtle bit of shading to denote box contents but never quite found the time to look in more detail!

**Reply by rmscode** [reply]
Understandable! It's been a while since I messed around with any css, but if I'm feeling ambitious and come up with something half decent, I'll let you know.

**Reply by EDIflyer** [reply]
@rmscode you might want to look at the solution by @alexvoss below - did the trick beautifully for me and the documentation from @squidfunk on adding extra CSS that he links to is spot on.  It took only a minute to add the `stylesheets\extra.css` files and amend `mkdocs.yml` 🚀

**Reply by rmscode** [reply]
Beautiful! I did see his reply, but I was away this past week and didn't have a chance to try it out. Thanks for following up with your experience though!

**Comment by alexvoss** [comment, marked as answer]
I am putting a box around tabbed environments for the same reason using this [extra CSS](https://squidfunk.github.io/mkdocs-material/customization#additional-css):

```CSS
.md-typeset .tabbed-set {
  border: 0.5px solid lightgray;
}

.md-typeset .tabbed-content {
  padding-left: 1rem;
  padding-right: 1rem;
}
```

Note that I am not we web designer, so this can probably be improved. The result looks like this:
![image](https://github.com/squidfunk/mkdocs-material/assets/4134224/756079ed-5e28-42e7-9d61-463471e4d8a7)

**Reply by EDIflyer** [reply]
Superb work @alexvoss 🥳

Before - not clear if the 'run frequency' section is part of it or not:
![image](https://github.com/squidfunk/mkdocs-material/assets/13610277/f1787259-11e3-4313-9092-49624b7543c4)

After - crystal clear:
![image](https://github.com/squidfunk/mkdocs-material/assets/13610277/d349f877-9bb9-4e7c-90bc-0ac19a5a4caa)

Very subtle but does what is needed I'd say 😃

**Reply by hellt** [reply]
Thanks @alexvoss for this example, works great.

I have one question, what color variable should one set to make sure that the table border uses the appropriate color when switched to the slate theme?


<img width="856" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/5679861/af34e12d-af46-49fd-ac0f-70a61c70a81e">

**Reply by hellt** [reply]
Ok, found it eventually

```css
.md-typeset .tabbed-set {
  border: 0.5px solid var(--md-typeset-table-color);
}
```

<img width="719" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/5679861/9863c15a-f11c-4f2b-8fcb-1fa95ec84bd3">
